src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildData.java

```diff
@@ -193,6 +195,18 @@ public BuildData(Run<?, ?> run, @Nullable TaskListener listener) throws IOExcept
             this.result = null;
             this.isCompleted = false;
         }
+        this.isBuilding = run.isBuilding();
+
+        // Set StartTime, EndTime and Duration
+        this.startTime = run.getStartTimeInMillis();
+        long durationInMs = run.getDuration();
+        if (durationInMs == 0 && startTime != 0) {
+            durationInMs = System.currentTimeMillis() - startTime;
+        }
+        this.duration = durationInMs;
+        if (duration != 0 && startTime != 0) {
+            this.endTime = startTime + duration;
+        }
```

Comment:
As we discussed offline, the `duration` and `endTime` would be omitted if the pipeline is running in the backend. Could we add a comment here just mentioning this? It might be confusing if you think that the endTime is sent for real purpose when the pipeline is running.

Reply:
Added the comment